### PR TITLE
[l10n] Improve Spanish (es-ES) locale

### PR DESCRIPTION
**Pull request by Macampu420**
- [x] Added missing spanish strings

- [x] Existent strings were capitalized based on french translations (thanks to romgrk)

- [X] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/grid/x-data-grid/src/locales/esES.ts [hunk 1]
```diff
@@ -71,33 +71,33 @@ const esESGrid: Partial<GridLocaleText> = {
...
-  // 'headerFilterOperator<': 'Less than',
-  // 'headerFilterOperator<=': 'Less than or equal to',
+  headerFilterOperatorContains: 'Contiene',
+  headerFilterOperatorEquals: 'Es igual a',
+  headerFilterOperatorStartsWith: 'Comienza con',
+  headerFilterOperatorEndsWith: 'Termina con',
+  headerFilterOperatorIs: 'Es',
+  headerFilterOperatorNot: 'No es',
+  headerFilterOperatorAfter: 'Esta despues de',
+  headerFilterOperatorOnOrAfter: 'Esta en o después de',
```
Comment (romgrk) [inline packages/grid/x-data-grid/src/locales/esES.ts:89]: Creo que le falta el accento al primero `despues` aqui.

File: packages/grid/x-data-grid/src/locales/esES.ts [hunk 2]
```diff
@@ -71,33 +71,33 @@ const esESGrid: Partial<GridLocaleText> = {
...
+  headerFilterOperatorOnOrBefore: 'Esta en o antes de',
+  headerFilterOperatorIsEmpty: 'Esta vacio',
+  headerFilterOperatorIsNotEmpty: 'No esta vacio',
+  headerFilterOperatorIsAnyOf: 'Es cualquiera de',
+  'headerFilterOperator=': 'es igual a',
+  'headerFilterOperator!=': 'es diferente a',
+  'headerFilterOperator>': 'es mayor que',
+  'headerFilterOperator>=': 'es mayor o igual que',
+  'headerFilterOperator<': 'es menor que',
+  'headerFilterOperator<=': 'es menor o igual que',
```
Comment (romgrk) [inline packages/grid/x-data-grid/src/locales/esES.ts:100]: I think these need to be capitalized like their English counterparts to be consistent.

Comment (Macampu420) [reply packages/grid/x-data-grid/src/locales/esES.ts:100]: Yep bro, that's right, thanks for feedback, already corrected!

File: packages/grid/x-data-grid/src/locales/esES.ts [hunk 3]
```diff
@@ -58,46 +58,46 @@ const esESGrid: Partial<GridLocaleText> = {
...
+  filterOperatorEndsWith: 'Termina con',
+  filterOperatorIs: 'Es',
+  filterOperatorNot: 'No es',
+  filterOperatorAfter: 'Es posterior',
+  filterOperatorOnOrAfter: 'Es en o posterior',
+  filterOperatorBefore: 'Es anterior',
+  filterOperatorOnOrBefore: 'Es en o anterior',
+  filterOperatorIsEmpty: 'Está vacío',
+  filterOperatorIsNotEmpty: 'No esta vacío',
+  filterOperatorIsAnyOf: 'Es cualquiera de',
```
Comment (romgrk) [inline packages/grid/x-data-grid/src/locales/esES.ts:73]: All lowercase, e.g. `es cualquiera de`

File: packages/grid/x-data-grid/src/locales/esES.ts [hunk 4]
```diff
@@ -58,46 +58,46 @@ const esESGrid: Partial<GridLocaleText> = {
...
+  'filterOperator=': 'es igual a',
+  'filterOperator!=': 'es diferente a',
+  'filterOperator>': 'es mayor que',
+  'filterOperator>=': 'es mayor o igual que',
+  'filterOperator<': 'es menor que',
+  'filterOperator<=': 'es menor o igual que',
```
Comment (romgrk) [inline packages/grid/x-data-grid/src/locales/esES.ts:79]: From what I can see, these need to stay in mathematical notation, there is nothing to translate here. Not sure but that's what I see elsewhere.

File: packages/grid/x-data-grid/src/locales/esES.ts (hunk 2 above)
Comment (romgrk) [reply packages/grid/x-data-grid/src/locales/esES.ts:100]: @Macampu420 like this:

image

**Review by romgrk — APPROVED** [review]
LGTM, thanks!

**Pull request merged**